src/operator.rs

```diff
@@ -931,6 +952,9 @@ mod tests {
                 "length".to_string(),
             ],
             None,
+            ParseOptions {
+                drop_nonmatching: true,
```

Comment:
as evidenced by getting it backwards, this name is kind of confusing. Lets just call it `no_drop` to be parallel with `lang`

Reply:
> as evidenced by getting it backwards, this name is kind of confusing. Lets just call it `no_drop` to be parallel with `lang`

I was actually going for the "don't have negated variable/settings names" rule - let me know if you still want to change it after the other changes, it's a bit clearer now.

I might add unit tests in typecheck.ts as well, as that would have caught this brainfart before the integration test.

Reply:
I love the "don't have negated variable names" rule!